### DISCUSSION TITLE
Transcoded, fixing corrupted frames, and re-encoded H264 to multiple target protocol

### DISCUSSION THREAD

**Discussion by JasonTan1036** [bluenviron/mediamtx#2861] [Questions]
### Question

Hi there, I am pretty new to this framework.  Supposedly, I want to stabilize, fixing corrupted frames, transcode, and muxed into H264.  Is this mediamtx server is capable of handling like this following scenario?
- from source rtsp (encoded: any) --> rtsp (re-encoded H264) --> then muxed to HLS (H264)
- from source rtsp (encoded: any) --> HLS (re-encoded H264)
- from source rtsp (encoded: any) --> rtmp (re-encoded H264) --> then muxed to HLS (H264)
- from source rtsp (encoded: any) --> rtmps (re-encoded H264) --> then muxed to HLS (H264)
- from source rtmp (encoded: any) --> rtmp (re-encoded H264) --> then muxed to HLS (H264)
- from source rtmps (encoded: any) -->rtmp (re-encoded H264)--> then muxed to HLS (H264)
- from source hls (encoded: any) --> hls (re-encoded H264)

**Comment by aler9** [comment]
Hello,

Remuxing (change of protocol without re-encoding) is automatic, you can publish with any supported protocol and read the same stream with another protocol. For instance, you can import a stream from a RTSP source and read it with RTMP, RTMPS, HLS or anything else. Instruction for reading from the server are in the [README](https://github.com/bluenviron/mediamtx?tab=readme-ov-file#read-from-the-server).

Re-encoding can be performed by integrating the server with FFmpeg or GStreamer. For instance, you can import a stream into a certain path and republish its re-encoded version inside another path. Instructions are in the [README](https://github.com/bluenviron/mediamtx?tab=readme-ov-file#remuxing-re-encoding-compression).

**Reply by OIZid** [reply]
How to reencode with GStreamer? I'm trying to reencode to AV1 and send through RTSP, which is apparently not possible with ffmpeg. In the readme, it implies it's possible to use GStreamer on the hooks, but it doesn't seem to be included with MediaMTX like ffmpeg is.

**Reply by RedShift1** [reply]
ffmpeg does not come with mediamtx, it's part of your linux distro (or your own separate download). You can use ffmpeg to reencode to AV1 and send it to mediamtx. See https://trac.ffmpeg.org/wiki/Encode/AV1 for how to encode to AV1.

**Reply by OIZid** [reply]
I meant ffmpeg is part (or can be) of the docker image when you get "bluenviron/mediamtx:latest-ffmpeg", but there is no such option for GStreamer. Also, ffmpeg does not allow you to stream AV1 through RTSP as far as I'm aware.

**Reply by RedShift1** [reply]
Try using SRT instead to publish to mediamtx?

**Reply by aler9** [reply]
```
rtspsrc location=rtsp://existingstream protocols=tcp latency=0 \
! rtph264depay \
svtav1enc target-bitrate=3000 max-bitrate=3000 preset=8 \
! av1parse \
! rtspclientsink location=rtspclientsink location=rtsp://localhost:8554/target-path
```
